---
title: "Microbes need frenemies: Mid-Cayman Rise"
format: html
editor: visual
---

# Set up environment - Mid-Cayman Rise

```{r}
# | message: false
library(tidyverse)
# library(tximport)
library(DESeq2)
```

Import MCR data from TXI subset.

```{r}
# | eval: false
load(file = "/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/txi-mcr.RData", verbose = TRUE)
# class(txi_mcr_euk_annot)
```

## Scaled data frame

```{r}
# | eval: false
mcr_counts_scaled <- makeCountsFromAbundance(
  as.matrix(txi_mcr_euk_annot$counts),
  as.matrix(txi_mcr_euk_annot$abundance),
  as.matrix(txi_mcr_euk_annot$length),
  countsFromAbundance = "scaledTPM")
# class(mcr_counts_scaled)
# head(mcr_counts_scaled)
mcr_df_scaled <- as.data.frame(mcr_counts_scaled)
```


## Count table

Strip column names of replicate information. Ready column headers for averaging across replicates.

```{r}
# | eval: false
names_orig <- colnames(mcr_df_scaled)
# names_orig
names_new_0 <- sub("_[^_]+$", "", names_orig)
# names_new_0
names_new_1 <- sub("MCR_\\d\\d_", "", names_new_0)
# names_new_1
colnames(mcr_df_scaled) <- names_new_1
```


```{r}
# | eval: false
mcr_mean_counts_df <- mcr_df_scaled %>%
  cbind(as.list(.) %>%
    Filter(is.numeric, .) %>% 
    split(names(.)) %>%
    lapply(as.data.frame) %>%
    lapply(rowMeans) %>%
    setNames(paste0("mean.", names(.)))) %>% 
  select(starts_with("mean"))

# head(mcr_mean_counts_df)
```

# Taxonomic breakdown

Import taxonomic and functional information.

```{r}
# | eval: false
taxfxn <- read_delim(file = "/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/TaxonomicAndFunctionalAnnotations.csv") #
head(taxfxn)[1:3,]
```

Join with mean of the transcript counts

```{r}
# | eval: false
# head(mcr_mean_counts_df)[1:3,]
mcr_annotated_mean <- dplyr::left_join(mcr_mean_counts_df %>%
                   mutate(SequenceID = rownames(mcr_mean_counts_df)),
                                 taxfxn,
                                 by = "SequenceID")
```

Isolate only taxonomic IDs

```{r}
# | eval: false
mcr_df_for_tax <- mcr_annotated_mean %>% 
  select(full_classification, starts_with("mean"), SequenceID) %>% 
  distinct()

mcr_wtax_only <- mcr_df_for_tax %>% 
  select(-SequenceID) %>% 
  pivot_longer(cols = starts_with("mean"), names_to = "SAMPLE", values_to = "scaledTPM") %>% 
  group_by(SAMPLE, full_classification) %>% 
    summarise(SUM_scaledTPM = sum(scaledTPM)) 
```

```{r}
# save(mcr_mean_counts_df, mcr_wtax_only, file = "/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_tax_means.RData")
```

## Import MCR taxonomic data

Pick up here locally!

```{r}
# load("/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_tax_means.RData", verbose = TRUE)
load("input-docs/mcr_df_tax_means.RData", verbose = TRUE)
```

Taxonomic breakdown of metatranscriptome data from Mid-Cayman Rise. First get taxonomic dataframe to compare to what we already know.

```{r}
taxa <- mcr_wtax_only %>%
  select(full_classification) %>% distinct() %>% 
  separate(full_classification, c("Domain", "Supergroup", "Phylum", "Class", "Order", "Family", "Genus_spp"), sep = "; ", remove = FALSE) %>% 
  mutate(
    SUPERGROUP = case_when(Supergroup == NA ~ "Eukaryote unannotated",
                           Supergroup == "Eukaryota incertae sedis" ~ "Eukaryote unannotated",
                           TRUE ~ Supergroup))
```

Modify supergroup level annotations.

```{r}
mcr_wtax_supergroup <- mcr_wtax_only %>% 
  separate(SAMPLE, c("FIELD", "LOCATION", "EXP", "SAMPLE_ID"), sep = "_") %>% 
  filter(EXP == "insitu") %>% 
  separate(full_classification, c("Domain", "Supergroup", "Phylum", "Class", "Order", "Family", "Genus_spp"), sep = "; ", remove = FALSE) %>% 
  mutate(
    SUPERGROUP = case_when(is.na(Supergroup) ~ "Eukaryote unannotated",
                           Supergroup == "Eukaryota incertae sedis" ~ "Eukaryote unannotated",
                           TRUE ~ Supergroup),
    VENT_FIELD = case_when(grepl("Piccard", FIELD) ~ "Piccard",
                           grepl("VonDamm", FIELD) ~ "Von Damm")) %>% 
  group_by(VENT_FIELD, LOCATION, SUPERGROUP) %>% 
    summarise(MEAN = mean(SUM_scaledTPM))
# Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species
# head(mcr_wtax_supergroup)
```

```{r}
# table(mcr_wtax_supergroup$SUPERGROUP)
```

```{r}
mcr_wtax_supergroup %>% 
  ggplot(aes(x = LOCATION, y = MEAN, fill = SUPERGROUP)) +
  geom_bar(stat = "identity", position = "fill", color = "black") +
    theme_linedraw() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  facet_grid(cols = vars(VENT_FIELD), scales = "free_x", space = "free")
```

### Fit to tag-seq IDs

```{r}
all_taxa_metat_order <- c("Alveolata-Ciliophora", "Alveolata-Dinoflagellata", "Other Alveolata", "Amoebozoa", "Apusozoa", "Archaeplastida","Excavata", "Hacrobia", "Rhizaria-Cercozoa",  "Rhizaria-Radiolaria","Rhizaria",  "Stramenopiles", "Stramenopiles-Ochrophyta","Stramenopiles-Opalozoa;Sagenista",  "Opisthokonta", "Other-metaT only", "Unknown Eukaryota")
all_taxa_metat_color = c("#fa9fb5", "#c51b8a", "#1d91c0","#67000d", "#ef3b2c", "#ffffcc","#feb24c",  "#c7e9b4",  "#253494", "#9e9ac8","lightblue", "#238b45", "#54278f", "#bdbdbd", "#fee6ce", "#cb181d", "#000000")
# length(all_taxa_metat_color)
# length(all_taxa_metat_order)
```

```{r}
# alv_names <- c("Ciliophora", "Dinophyta", "Perkinsea", "Colponemidia", "Chromerida")
as_is <- c("Amoebozoa", "Apusozoa", "Excavata", "Hacrobia", "Archaeplastida")

taxa <- mcr_wtax_only %>%
  select(full_classification, -SAMPLE) %>% distinct() %>% 
  separate(full_classification, c("Domain", "Supergroup", "Phylum", "Class", "Order", "Family", "Genus_spp"), sep = "; ", remove = FALSE) %>% 
  mutate(SUPERGROUP_18S = case_when(
    Phylum == "Ciliophora" ~ "Alveolata-Ciliophora",
    Phylum == "Dinophyta" ~ "Alveolata-Dinoflagellata",
    # Phylum == "Perkinsea" ~ "Protalveolata",
    # Phylum == "Colponemidia" ~ "Protalveolata",
    # Phylum == "Chromerida" ~ "Protalveolata",
    Supergroup == "Alveolata" ~ "Other Alveolata",
    Supergroup %in% as_is ~ Supergroup,
    Supergroup == "Haptista" ~ "Hacrobia",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    (Supergroup == "Rhizaria" & Phylum != "Radiolaria" & Phylum != "Cercozoa") ~ "Rhizaria",
    Order == "Bigyra" ~ "Stramenopiles-Opalozoa;Sagenista",
    Class == "Ochromonadales" ~ "Stramenopiles-Ochrophyta",
    Supergroup == "Stramenopiles" ~ "Stramenopiles",
    Supergroup == "Opisthokonta" ~ "Opisthokonta",
    (is.na(Supergroup) | Supergroup == "Eukaryota incertae sedis") ~ "Unknown Eukaryota",
    TRUE ~ "Other-metaT only")) %>% 
  mutate(SUPERGROUP_18S_ORDER = factor(SUPERGROUP_18S, levels = all_taxa_metat_order))
# unique(taxa$SUPERGROUP_18S_ORDER)
taxa <- taxa %>% ungroup() %>% select(-SAMPLE) %>% distinct()
```

```{r}
mcr_wtax_only %>% 
  separate(SAMPLE, c("FIELD", "LOCATION", "EXP", "SAMPLE_ID"), sep = "_") %>% 
  filter(EXP == "insitu") %>% 
  separate(full_classification, c("Domain", "Supergroup", "Phylum", "Class", "Order", "Family", "Genus_spp"), sep = "; ", remove = FALSE) %>% 
  mutate(SUPERGROUP_18S = case_when(
    Phylum == "Ciliophora" ~ "Alveolata-Ciliophora",
    Phylum == "Dinophyta" ~ "Alveolata-Dinoflagellata",
    # Phylum == "Perkinsea" ~ "Protalveolata",
    # Phylum == "Colponemidia" ~ "Protalveolata",
    # Phylum == "Chromerida" ~ "Protalveolata",
    Supergroup == "Alveolata" ~ "Other Alveolata",
    Supergroup %in% as_is ~ Supergroup,
    Supergroup == "Haptista" ~ "Hacrobia",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    (Supergroup == "Rhizaria" & Phylum != "Radiolaria" & Phylum != "Cercozoa") ~ "Rhizaria",
    Order == "Bigyra" ~ "Stramenopiles-Opalozoa;Sagenista",
    Class == "Ochromonadales" ~ "Stramenopiles-Ochrophyta",
    Supergroup == "Stramenopiles" ~ "Stramenopiles",
    Supergroup == "Opisthokonta" ~ "Opisthokonta",
    (is.na(Supergroup) | Supergroup == "Eukaryota incertae sedis") ~ "Unknown Eukaryota",
    TRUE ~ "Other-metaT only")) %>% 
  mutate(VENT_FIELD = case_when(grepl("Piccard", FIELD) ~ "Piccard",
                           grepl("VonDamm", FIELD) ~ "Von Damm")) %>% 
  mutate(VENT_BIN = case_when(
    (LOCATION == "Background" | LOCATION == "Plume") ~ "Non-vent",
    TRUE ~ "Vent"
  )) %>% 
  group_by(VENT_FIELD, VENT_BIN, LOCATION, SUPERGROUP_18S) %>% 
    summarise(MEAN = mean(SUM_scaledTPM)) %>% 
  mutate(SUPERGROUP_18S_ORDER = factor(SUPERGROUP_18S, levels = all_taxa_metat_order)) %>% 
  ggplot(aes(x = LOCATION, y = MEAN, fill = SUPERGROUP_18S_ORDER)) +
  geom_bar(stat = "identity", position = "fill", color = "black") +
  scale_fill_manual(values = all_taxa_metat_color) +
    theme_linedraw() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1),
          strip.background = element_blank(),
          strip.text = element_text(color = "black")) +
  facet_grid(cols = vars(VENT_FIELD, VENT_BIN), scales = "free_x", space = "free")
```

# DE seq MCR

See scripts: \`metaT-deseq-Vent_NonVent.R'

Import transcripts processed through DESeq to look at differentially expressed genes based on if the sample was in the vent versus outside of the vent.

```{r, message=FALSE}
library(DESeq2); library(tidyverse)
```

```{r}
# | eval: false
taxfxn <- read_delim(file = "/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/TaxonomicAndFunctionalAnnotations.csv") #
# head(taxfxn)[1:3,]
```
## Import and reformat KEGG DB
```{r}
kegg <- read.csv("../KEGG_DB/combined_kegg.csv")
# dim(kegg)
curated_kegg <- read.csv("../KEGG_DB/reformat-kegg-pfam-skh.csv")

key_geneid <- curated_kegg %>% 
  select(-X) %>% 
  right_join(kegg %>% select(KEGG = KO_number, everything(), -X)) %>% 
  distinct() %>% 
  select(starts_with("KeggOrthology_"), Category01, Category02, FullName, GeneID, Gene_identification, KEGG, PFAM, Descriptions, REF = REFs)
```

```{r}
# | eval: false
load("/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr-insitu_vent_vs_novent_DESeq.RData", verbose = TRUE)
```


Tranform DEseq class information to a dataframe.

```{r}
# | eval: false
# class(de_mcr_annot)
results_mcr_annot <- results(de_mcr_annot, alpha = 0.05)
#Snapshot of results. 
# summary(results_mcr_annot)
```

```{r}
# | eval: false
mcr_transcripts <- data.frame(results_mcr_annot) %>% 
  mutate(REGULATION = case_when(
    log2FoldChange > 0 ~ "upregulated in vents",
    log2FoldChange < 0 ~ "upregulated in background"
  ),
  SIGNIFICANT = case_when(
    pvalue <= 0.05 ~ "Significantly",
    TRUE ~ "Not significantly"
  )) %>% 
  # filter(SIGNIFICANT == "Significantly") %>% 
  rownames_to_column(var = "SequenceID")
```




Load Rdata locally.

```{r}
# load("/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_DE_annot.RData", verbose = TRUE)
load("input-docs/mcr_df_DE_annot.RData", verbose = TRUE)
# load("/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_tax_means.RData", verbose = TRUE)
load("input-docs/mcr_df_tax_means.RData", verbose = TRUE)
```

```{r}
# glimpse(results_mcr_annot)
glimpse(mcr_transcripts)
```

## Plot log FC

```{r, fig.height=5, fig.width=6}
# head(mcr_transcripts)
data.frame(mcr_transcripts) %>% 
  ggplot(aes(x = baseMean, y = log2FoldChange, color = SIGNIFICANT)) +
  geom_point(stat = "identity", shape = 20, size = 2) +
  scale_x_log10() +
  theme_classic() +
  scale_color_manual(values = c("#878787", "#d73027")) +
  theme(legend.position = "top",
        axis.text = element_text(color = "black", size = 12, face = "bold"),
        axis.title = element_text(color = "black", size = 12, face = "bold")) +
  labs(y = "log2 Fold Change", x = "Mean Normalized Counts")

```


```{r, fig.height=5, fig.width=6}
# head(mcr_transcripts)
data.frame(mcr_trans_wKEGG) %>% 
  mutate(KEGG = str_remove(KEGG_ko, "ko:")) %>%
  left_join(key_geneid) %>% 
  filter(!is.na(Category01)) %>% 
  ggplot(aes(x = baseMean, y = log2FoldChange, color = Category01)) +
  geom_point(stat = "identity", shape = 20, size = 2) +
  scale_x_log10() +
  theme_classic() +
  # scale_color_manual(values = c("#878787", "#d73027")) +
  theme(legend.position = "top",
        axis.text = element_text(color = "black", size = 12, face = "bold"),
        axis.title = element_text(color = "black", size = 12, face = "bold")) +
  labs(y = "log2 Fold Change", x = "Mean Normalized Counts")

```

What are the highly significant and represented transcripts for both sides?

```{r}
upreg_10 <- mcr_trans_wKEGG %>% 
  filter(SIGNIFICANT == "Significantly") %>% 
    # filter(abs(log2FoldChange) > 10) %>% 
  mutate(KEGG = str_remove(KEGG_ko, "ko:")) %>%
  left_join(key_geneid) %>% 
  filter(!is.na(Category01)) %>% 
  left_join(taxa)
glimpse(upreg_10)
# unique(upreg_10$Category01)
# range(upreg_10$baseMean)
```

```{r, fig.height=5, fig.width=6}
pink_qual <- c("#fff7f3", "#fcc5c0", "#fa9fb5","#f768a1","#dd3497", "#ae017e", "#7a0177")
mcr_trans_wKEGG %>% 
  mutate(KEGG = str_remove(KEGG_ko, "ko:")) %>%
  left_join(key_geneid) %>% 
  left_join(taxa) %>% 
  filter(!is.na(Category01)) %>%
  filter(Category01 == "Phagotrophy") %>% 
  ggplot(aes(x = GeneID, y = SUPERGROUP_18S, fill = log2(baseMean), color = SIGNIFICANT)) +
  geom_tile(stat = "identity", size = 0.6) +
  scale_color_manual(values = c("black", "white")) +
      scale_fill_stepsn(
      # limits = c(-5,5),
      # breaks = c(0.0, 3.5, 7.0, 10.5),
      show.limits = TRUE,
      colors = pink_qual) +
  # coord_fixed(ratio = 1) +
  facet_grid(cols = vars(Category01), rows = vars(REGULATION)) + 
  theme_classic() +
  theme(axis.text = element_text(color = "black"),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 6),
        axis.title = element_blank())
```



```{r}
# | eval: false
# head(mcr_transcripts)
# colnames(taxfxn)

kegg_mod <- kegg %>% select(-X, KEGG = KO_number, everything())

mcr_trans_wKEGG <- mcr_transcripts %>% 
  filter(SIGNIFICANT == "Significantly") %>% 
  left_join(taxfxn) %>% 
  mutate(KEGG = str_remove_all(KEGG_ko, "ko:")) %>%
  mutate(KEGG = strsplit(as.character(KEGG), ",")) %>% 
    unnest(KEGG) %>% 
  left_join(kegg_mod)
```

```{r}
all_taxa_metat_order <- c("Alveolata-Ciliophora", "Alveolata-Dinoflagellata", "Other Alveolata", "Amoebozoa", "Apusozoa", "Archaeplastida","Excavata", "Hacrobia", "Rhizaria-Cercozoa",  "Rhizaria-Radiolaria","Rhizaria",  "Stramenopiles", "Stramenopiles-Ochrophyta","Stramenopiles-Opalozoa;Sagenista",  "Opisthokonta", "Other-metaT only", "Unknown Eukaryota")
all_taxa_metat_color = c("#fa9fb5", "#c51b8a", "#1d91c0","#67000d", "#ef3b2c", "#ffffcc","#feb24c",  "#c7e9b4",  "#253494", "#9e9ac8","lightblue", "#238b45", "#54278f", "#bdbdbd", "#fee6ce", "#cb181d", "#000000","#edf8fb", "#99d8c9", "#005824", "#41ae76", "#d9f0a3", "#fee391", "#fe9929", "#cc4c02", "#9ebcda","#8c6bb1","#88419d","#66c2a4", "#addd8e", "#238443","#ccece6", "#8c96c6","#fcbba1","#fc9272","#fb6a4a","#cb181d", "#d4b9da","#c994c7","#df65b0","#e7298a", "white", "black")




"#66c2a4"
"#238b45"
"#ccece6"
"#ffffcc"
"#addd8e"
"#78c679"
"#41ab5d"
"#238443"
"#005a32"
"#ffffd4"
"#fec44f"
"#fe9929"
"#ec7014"
"#cc4c02"
"#8c2d04"
"#edf8fb"
"#bfd3e6"
"#9ebcda"
"#8c96c6"
"#8c6bb1"
"#88419d"
"#6e016b"

"#fee5d9"
"#fcbba1"
"#fc9272"
"#fb6a4a"
"#ef3b2c"
"#cb181d"

"#d9d9d9"
"#bdbdbd"
"#969696"
"#737373"
"#525252"
"#252525"

"#f1eef6"
"#d4b9da"
"#c994c7"
"#df65b0"
"#e7298a"
"#ce1256"
"#91003f"
head(mcr_trans_wKEGG)
```

```{r}
glimpse(mcr_trans_wKEGG)
```


```{r, fig.height=8, fig.width=4}
# library(ggdist)
mcr_trans_wKEGG %>%
  filter(!is.na(KeggOrthology_A)) %>% 
  ggplot(aes(y = log2FoldChange, fill = KeggOrthology_A)) +
  stat_slab(aes(thickness = after_stat(pdf*n)), scale = 0.7) +
  stat_dotsinterval(side = "bottom", scale = 0.7, slab_linewidth = NA) +
  # ggdist::geom_dots(side = "both") +
  # ggdist::stat_dotsinterval() +
  coord_flip() +
  # geom_hline(yintercept = 0) +
  scale_fill_manual(values = all_taxa_metat_color) +
  scale_color_manual(values = all_taxa_metat_color) +
  theme_classic() +
  facet_grid(rows = vars(KeggOrthology_A), scales = "free") +
  theme(legend.position = "top") +
  labs(y = "Transcript sum", x = "")
```


```{r, fig.height=6, fig.width=7}
# head(mcr_trans_wKEGG)
# svg("plot-prelim-barplot.svg", h = 6, w = 7)
# glimpse(mcr_trans_wKEGG)
mcr_toplot <- mcr_trans_wKEGG %>% 
  mutate(KEGG = str_remove(KEGG_ko, "ko:")) %>%
  left_join(kegg)
  select(starts_with("Kegg"), baseMean, log2FoldChange, REGULATION, SIGNIFICANT, KEGG, A:D, full_classification)
dim(mcr_toplot)
mcr_toplot %>%
  group_by(KeggOrthology_A, REGULATION) %>% 
    reframe(SUM = log2FoldChange, .groups = "rowwise") %>% 
  select(LEVEL_NAME = KeggOrthology_A, SUM, REGULATION) %>%
  add_column(CATEGORY = "Kegg level A") #%>% 

  distinct() %>% 
  ungroup() %>% 
  add_row(mcr_toplot %>% ungroup() %>% 
    group_by(KeggOrthology_B, REGULATION) %>% 
      reframe(.by = SUM = log2FoldChange) %>% 
    select(LEVEL_NAME = KeggOrthology_B, SUM, REGULATION) %>%
    add_column(CATEGORY = "Kegg level B")) %>% 
  ungroup() %>% 
  add_row(mcr_toplot %>% ungroup() %>% 
            left_join(taxa) %>% 
            group_by(SUPERGROUP_18S, REGULATION) %>% 
              reframe(SUM = as.numeric(log2FoldChange)) %>% 
            select(LEVEL_NAME = SUPERGROUP_18S, SUM, REGULATION) %>% 
            add_column(CATEGORY = "Supergroup")) %>% 
  distinct() %>% 
  ggplot(aes(x = CATEGORY, y = SUM, fill = LEVEL_NAME)) +
    geom_bar(stat = "identity", position = "stack", width = 0.4) +
   coord_flip() +
  geom_hline(yintercept = 0) +
  scale_fill_manual(values = all_taxa_metat_color) +
  theme_classic() +
  facet_grid(rows = vars(CATEGORY), scales = "free") +
  theme(legend.position = "top",
        strip.text = element_blank(),
        axis.line.y = element_blank()) +
  labs(y = "Transcript sum", x = "")
# dev.off()
```

```{r}
save(mcr_trans_wKEGG, mcr_transcripts, file = "/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_DE_annot.RData")
```



# MCR a look a specific taxa

```{r}
# load("/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_DE_annot.RData", verbose = TRUE)
load("input-docs/mcr_df_DE_annot.RData", verbose = TRUE)
# load("/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_tax_means.RData", verbose = TRUE)
load("input-docs/mcr_df_tax_means.RData", verbose = TRUE)
# dim(mcr_trans_wKEGG)
dim(mcr_transcripts)
head(mcr_transcripts)
```

# Session Information

```{r}
sessionInfo()
```
