---
title: "Microbes need frenemies: Mid-Cayman Rise"
format: html
editor: visual
---

# Set up environment - Mid-Cayman Rise

```{r}
library(tidyverse)
library(tximport)
```

Import MCR data from TXI subset.

```{r}
load(file = "/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/txi-mcr.RData", verbose = TRUE)
# class(txi_mcr_euk_annot)
```

## Scaled data frame

```{r}
mcr_counts_scaled <- makeCountsFromAbundance(
  as.matrix(txi_mcr_euk_annot$counts),
  as.matrix(txi_mcr_euk_annot$abundance),
  as.matrix(txi_mcr_euk_annot$length),
  countsFromAbundance = "scaledTPM")
# class(mcr_counts_scaled)
# head(mcr_counts_scaled)
mcr_df_scaled <- as.data.frame(mcr_counts_scaled)
```

```{r}
# dim(mcr_df_scaled)
# head(mcr_df_scaled)
```

## Count table

Strip column names of replicate information. Ready column headers for averaging across replicates.

```{r}
names_orig <- colnames(mcr_df_scaled)
# names_orig
names_new_0 <- sub("_[^_]+$", "", names_orig)
# names_new_0
names_new_1 <- sub("MCR_\\d\\d_", "", names_new_0)
# names_new_1
colnames(mcr_df_scaled) <- names_new_1
```

```{r}
# head(mcr_df_scaled)
```

```{r}
mcr_mean_counts_df <- mcr_df_scaled %>%
  cbind(as.list(.) %>%
    Filter(is.numeric, .) %>% 
    split(names(.)) %>%
    lapply(as.data.frame) %>%
    lapply(rowMeans) %>%
    setNames(paste0("mean.", names(.)))) %>% 
  select(starts_with("mean"))

# head(mcr_mean_counts_df)
```

# Taxonomic breakdown

Import taxonomic and functional information.

```{r}
taxfxn <- read_delim(file = "/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/TaxonomicAndFunctionalAnnotations.csv") #
head(taxfxn)[1:3,]
```

Join with mean of the transcript counts

```{r}
# head(mcr_mean_counts_df)[1:3,]
mcr_annotated_mean <- dplyr::left_join(mcr_mean_counts_df %>%
                   mutate(SequenceID = rownames(mcr_mean_counts_df)),
                                 taxfxn,
                                 by = "SequenceID")
```

Isolate only taxonomic IDs

```{r}
mcr_df_for_tax <- mcr_annotated_mean %>% 
  select(full_classification, starts_with("mean"), SequenceID) %>% 
  distinct()

mcr_wtax_only <- mcr_df_for_tax %>% 
  select(-SequenceID) %>% 
  pivot_longer(cols = starts_with("mean"), names_to = "SAMPLE", values_to = "scaledTPM") %>% 
  group_by(SAMPLE, full_classification) %>% 
    summarise(SUM_scaledTPM = sum(scaledTPM)) 
```

```{r}
# save(mcr_mean_counts_df, mcr_wtax_only, file = "/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_tax_means.RData")
```

## Import MCR taxonomic data

Pick up here locally!

```{r}
load("/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_tax_means.RData", verbose = TRUE)
# load("mcr_df_tax_means.RData", verbose = TRUE)
```

```{r}
head(mcr_wtax_only)
```

Taxonomic breakdown of metatranscriptome data from Mid-Cayman Rise.

```{r}
taxa <- mcr_wtax_only %>%
  select(full_classification) %>% distinct() %>% 
  separate(full_classification, c("Domain", "Supergroup", "Phylum", "Class", "Order", "Family", "Genus_spp"), sep = "; ", remove = FALSE) %>% 
  mutate(
    SUPERGROUP = case_when(Supergroup == NA ~ "Eukaryote unannotated",
                           Supergroup == "Eukaryota incertae sedis" ~ "Eukaryote unannotated",
                           TRUE ~ Supergroup))
  
# head(taxa)
# unique(taxa$SUPERGROUP)
# head(mcr_wtax_only)
```

```{r}
# head(mcr_wtax_only)
mcr_wtax_supergroup <- mcr_wtax_only %>% 
  separate(SAMPLE, c("FIELD", "LOCATION", "EXP", "SAMPLE_ID"), sep = "_") %>% 
  filter(EXP == "insitu") %>% 
  separate(full_classification, c("Domain", "Supergroup", "Phylum", "Class", "Order", "Family", "Genus_spp"), sep = "; ", remove = FALSE) %>% 
  mutate(
    SUPERGROUP = case_when(is.na(Supergroup) ~ "Eukaryote unannotated",
                           Supergroup == "Eukaryota incertae sedis" ~ "Eukaryote unannotated",
                           TRUE ~ Supergroup),
    VENT_FIELD = case_when(grepl("Piccard", FIELD) ~ "Piccard",
                           grepl("VonDamm", FIELD) ~ "Von Damm")) %>% 
  group_by(VENT_FIELD, LOCATION, SUPERGROUP) %>% 
    summarise(MEAN = mean(SUM_scaledTPM))
# Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species
# head(mcr_wtax_supergroup)
```

```{r}
# table(mcr_wtax_supergroup$SUPERGROUP)
```

```{r}
mcr_wtax_supergroup %>% 
  ggplot(aes(x = LOCATION, y = MEAN, fill = SUPERGROUP)) +
  geom_bar(stat = "identity", position = "fill", color = "black") +
    theme_linedraw() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  facet_grid(cols = vars(VENT_FIELD), scales = "free_x", space = "free")
```

### Fit to tag-seq IDs

```{r}
# all_taxa_order <- c("Alveolata-Ciliophora", "Alveolata-Dinoflagellata", "Protalveolata", "Other Alveolata", "Amoebozoa", "Apusozoa", "Excavata", "Hacrobia", "Archaeplastida", "Rhizaria", "Rhizaria-Radiolaria", "Rhizaria-Cercozoa", "Stramenopiles", "Stramenopiles-Opalozoa", "Stramenopiles-Sagenista", "Stramenopiles-Ochrophyta", "Opisthokonta", "Unknown Eukaryota")
# length(all_taxa_order)
# "Stramenopiles-Opalozoa;Sagenista"
# "Other-metaT only"

# all_taxa_color = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c","#cc4c02", "#c7e9b4", "#238b45", "#c6dbef", "#99d8c9","#1d91c0", "#253494", "#9e9ac8", "#54278f", "#bdbdbd", "#252525", "#f0f0f0")

all_taxa_metat_order <- c("Alveolata-Ciliophora", "Alveolata-Dinoflagellata", "Other Alveolata", "Amoebozoa", "Apusozoa", "Archaeplastida","Excavata", "Hacrobia", "Rhizaria-Cercozoa",  "Rhizaria-Radiolaria","Rhizaria",  "Stramenopiles", "Stramenopiles-Ochrophyta","Stramenopiles-Opalozoa;Sagenista",  "Opisthokonta", "Other-metaT only", "Unknown Eukaryota")
all_taxa_metat_color = c("#fa9fb5", "#c51b8a", "#1d91c0","#67000d", "#ef3b2c", "#ffffcc","#feb24c",  "#c7e9b4",  "#253494", "#9e9ac8","lightblue", "#238b45", "#54278f", "#bdbdbd", "#fee6ce", "#cb181d", "#000000")
# length(all_taxa_metat_color)
# length(all_taxa_metat_order)
```

```{r}
# alv_names <- c("Ciliophora", "Dinophyta", "Perkinsea", "Colponemidia", "Chromerida")
as_is <- c("Amoebozoa", "Apusozoa", "Excavata", "Hacrobia", "Archaeplastida")

taxa <- mcr_wtax_only %>%
  select(full_classification) %>% distinct() %>% 
  separate(full_classification, c("Domain", "Supergroup", "Phylum", "Class", "Order", "Family", "Genus_spp"), sep = "; ", remove = FALSE) %>% 
  mutate(SUPERGROUP_18S = case_when(
    Phylum == "Ciliophora" ~ "Alveolata-Ciliophora",
    Phylum == "Dinophyta" ~ "Alveolata-Dinoflagellata",
    # Phylum == "Perkinsea" ~ "Protalveolata",
    # Phylum == "Colponemidia" ~ "Protalveolata",
    # Phylum == "Chromerida" ~ "Protalveolata",
    Supergroup == "Alveolata" ~ "Other Alveolata",
    Supergroup %in% as_is ~ Supergroup,
    Supergroup == "Haptista" ~ "Hacrobia",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    (Supergroup == "Rhizaria" & Phylum != "Radiolaria" & Phylum != "Cercozoa") ~ "Rhizaria",
    Order == "Bigyra" ~ "Stramenopiles-Opalozoa;Sagenista",
    Class == "Ochromonadales" ~ "Stramenopiles-Ochrophyta",
    Supergroup == "Stramenopiles" ~ "Stramenopiles",
    Supergroup == "Opisthokonta" ~ "Opisthokonta",
    (is.na(Supergroup) | Supergroup == "Eukaryota incertae sedis") ~ "Unknown Eukaryota",
    TRUE ~ "Other-metaT only")) %>% 
  mutate(SUPERGROUP_18S_ORDER = factor(SUPERGROUP_18S, levels = all_taxa_metat_order))
# unique(taxa$SUPERGROUP_18S_ORDER)
```

```{r}
mcr_wtax_only %>% 
  separate(SAMPLE, c("FIELD", "LOCATION", "EXP", "SAMPLE_ID"), sep = "_") %>% 
  filter(EXP == "insitu") %>% 
  separate(full_classification, c("Domain", "Supergroup", "Phylum", "Class", "Order", "Family", "Genus_spp"), sep = "; ", remove = FALSE) %>% 
  mutate(SUPERGROUP_18S = case_when(
    Phylum == "Ciliophora" ~ "Alveolata-Ciliophora",
    Phylum == "Dinophyta" ~ "Alveolata-Dinoflagellata",
    # Phylum == "Perkinsea" ~ "Protalveolata",
    # Phylum == "Colponemidia" ~ "Protalveolata",
    # Phylum == "Chromerida" ~ "Protalveolata",
    Supergroup == "Alveolata" ~ "Other Alveolata",
    Supergroup %in% as_is ~ Supergroup,
    Supergroup == "Haptista" ~ "Hacrobia",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    (Supergroup == "Rhizaria" & Phylum != "Radiolaria" & Phylum != "Cercozoa") ~ "Rhizaria",
    Order == "Bigyra" ~ "Stramenopiles-Opalozoa;Sagenista",
    Class == "Ochromonadales" ~ "Stramenopiles-Ochrophyta",
    Supergroup == "Stramenopiles" ~ "Stramenopiles",
    Supergroup == "Opisthokonta" ~ "Opisthokonta",
    (is.na(Supergroup) | Supergroup == "Eukaryota incertae sedis") ~ "Unknown Eukaryota",
    TRUE ~ "Other-metaT only")) %>% 
  mutate(VENT_FIELD = case_when(grepl("Piccard", FIELD) ~ "Piccard",
                           grepl("VonDamm", FIELD) ~ "Von Damm")) %>% 
  mutate(VENT_BIN = case_when(
    (LOCATION == "Background" | LOCATION == "Plume") ~ "Non-vent",
    TRUE ~ "Vent"
  )) %>% 
  group_by(VENT_FIELD, VENT_BIN, LOCATION, SUPERGROUP_18S) %>% 
    summarise(MEAN = mean(SUM_scaledTPM)) %>% 
  mutate(SUPERGROUP_18S_ORDER = factor(SUPERGROUP_18S, levels = all_taxa_metat_order)) %>% 
  ggplot(aes(x = LOCATION, y = MEAN, fill = SUPERGROUP_18S_ORDER)) +
  geom_bar(stat = "identity", position = "fill", color = "black") +
  scale_fill_manual(values = all_taxa_metat_color) +
    theme_linedraw() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  facet_grid(cols = vars(VENT_FIELD, VENT_BIN), scales = "free_x", space = "free")
```

# DE seq MCR

See scripts: \`metaT-deseq-Vent_NonVent.R'

Import transcripts processed through DESeq to look at differentially expressed genes based on if the sample was in the vent versus outside of the vent.

```{r, message=FALSE}
library(DESeq2); library(tidyverse)
```

```{r}
taxfxn <- read_delim(file = "/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/TaxonomicAndFunctionalAnnotations.csv") #
head(taxfxn)[1:3,]
```
```{r}
kegg <- read.csv("../KEGG_DB/combined_kegg.csv")
# dim(kegg)
curated_kegg <- read.csv("../KEGG_DB/reformat-kegg-pfam-skh.csv")
```


```{r}
load("/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr-insitu_vent_vs_novent_DESeq.RData", verbose = TRUE)
```

Tranform DEseq class information to a dataframe. 

```{r}
class(de_mcr_annot)
results_mcr_annot <- results(de_mcr_annot, alpha = 0.05)
#Snapshot of results. 
summary(results_mcr_annot)
```


```{r}
mcr_transcripts <- data.frame(results_mcr_annot) %>% 
  mutate(REGULATION = case_when(
    log2FoldChange > 0 ~ "upregulated in vents",
    log2FoldChange < 0 ~ "upregulated in background"
  ),
  SIGNIFICANT = case_when(
    pvalue <= 0.05 ~ "Significantly",
    TRUE ~ "Not significantly"
  )) %>% 
  # filter(SIGNIFICANT == "Significantly") %>% 
  rownames_to_column(var = "SequenceID")
```

```{r}
glimpse(results_mcr_annot)
glimpse(mcr_transcripts)
```

```{r}
# head(mcr_transcripts)
data.frame(mcr_transcripts) %>% 
  ggplot(aes(x = baseMean, y = log2FoldChange, color = SIGNIFICANT)) +
  geom_point(stat = "identity") +
  scale_x_log10() +
  theme_classic() +
  scale_color_manual(values = c("#878787", "#d73027"))
# +
  # labs(title = mcols(res_hapto)$description[2])
```


```{r}
# head(mcr_transcripts)
# colnames(taxfxn)

kegg_mod <- kegg %>% select(-X, KEGG = KO_number, everything())

mcr_trans_wKEGG <- mcr_transcripts %>% 
  filter(SIGNIFICANT == "Significantly") %>% 
  left_join(taxfxn) %>% 
  mutate(KEGG = str_remove_all(KEGG_ko, "ko:")) %>%
  mutate(KEGG = strsplit(as.character(KEGG), ",")) %>% 
    unnest(KEGG) %>% 
  left_join(kegg_mod)
```

```{r}
head(mcr_trans_wKEGG)
mcr_trans_wKEGG %>% 
  group_by(KeggOrthology_A, REGULATION) %>% 
  summarise(SUM = log2FoldChange) %>% 
  ggplot(aes(x = REGULATION, y = SUM)) + 
    geom_bar(stat = "identity")
```

```{r}
save(mcr_trans_wKEGG, mcr_transcripts, file = "/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_DE_annot.RData")
```



# MCR a look a specific taxa
```{r}
load("/scratch/group/hu-lab/frenemies/euk-metaT-eukrhythmic-output/mcr_df_tax_means.RData", verbose = TRUE)

```


# Session Information

```{r}
sessionInfo()
```
